pkg/accountmanager/poller_test.go

```diff
@@ -0,0 +1,379 @@
+					VMSelector:  []v1alpha1.VirtualMachineSelector{},
+				},
+			}
+
+			// Delete selector.
+			accountPollerObj.addOrUpdateSelector(selector)
+			Expect(len(accountPollerObj.vmSelector.List())).To(Equal(0))
+		})
+		It("Remove selector", func() {
+			vmLabelsMap := map[string]string{
+				labels.CloudAccountName:      testAccountNamespacedName.Name,
+				labels.CloudAccountNamespace: testAccountNamespacedName.Namespace,
+				labels.CloudVmUID:            "ubuntu",
+				labels.CloudVpcUID:           "vpcid",
+				labels.VpcName:               "vpc",
+			}
+			vmList := make(map[string]*runtimev1alpha1.VirtualMachine)
+			vmObj := new(runtimev1alpha1.VirtualMachine)
+			vmObj.Name = "ubuntu"
+			vmObj.Namespace = testAccountNamespacedName.Namespace
+			vmObj.Labels = vmLabelsMap
+			vmList["ubuntu"] = vmObj
+			accountPollerObj.inventory.BuildVmCache(vmList, &testAccountNamespacedName)
+			vms := accountPollerObj.inventory.GetAllVms()
+			Expect(len(vms)).To(Equal(1))
+
+			// Delete VMs from cache.
+			accountPollerObj.removeSelector(&testAccountNamespacedName)
+			vms = accountPollerObj.inventory.GetAllVms()
+			Expect(len(vms)).To(Equal(0))
+		})
+		It("Account polling", func() {
+			vpcLabelsMap := map[string]string{
+				labels.CloudAccountName:      testAccountNamespacedName.Name,
+				labels.CloudAccountNamespace: testAccountNamespacedName.Namespace,
+				labels.CloudRegion:           "west",
+			}
+			vpcList := make(map[string]*runtimev1alpha1.Vpc)
+			vpcObj1 := new(runtimev1alpha1.Vpc)
+			vpcObj1.Name = "obj1"
+			vpcObj1.Namespace = testAccountNamespacedName.Namespace
+			vpcObj1.Labels = vpcLabelsMap
+			vpcObj1.Status.CloudId = "vpcid"
+			vpcObj1.Status.CloudName = "vpc"
+			vpcList["vpcid"] = vpcObj1
+
+			_ = fakeClient.Create(context.Background(), secret)
+			_ = fakeClient.Create(context.Background(), account)
+
+			accountPollerObj.namespacedName = &testAccountNamespacedName
+			mockCloudInterface.EXPECT().DoInventoryPoll(&testAccountNamespacedName).Return(nil).AnyTimes()
+			mockCloudInterface.EXPECT().GetAccountStatus(&testAccountNamespacedName).Return(&v1alpha1.
+				CloudProviderAccountStatus{}, nil).AnyTimes()
+
+			// Invalid VPC.
+			mockCloudInterface.EXPECT().GetVpcInventory(&testAccountNamespacedName).Return(nil,
+				fmt.Errorf("error")).Times(1)
+			accountPollerObj.doAccountPolling()
+			Expect(len(accountPollerObj.inventory.GetAllVpcs())).To(Equal(0))
+
+			// Empty VPC.
+			mockCloudInterface.EXPECT().GetVpcInventory(&testAccountNamespacedName).Return(nil,
+				nil).Times(1)
+			accountPollerObj.doAccountPolling()
+			Expect(len(accountPollerObj.inventory.GetAllVpcs())).To(Equal(0))
+
+			// Valid VPC.
+			mockCloudInterface.EXPECT().GetVpcInventory(&testAccountNamespacedName).Return(vpcList,
+				nil).Times(1)
+			accountPollerObj.doAccountPolling()
+			Expect(len(accountPollerObj.inventory.GetAllVpcs())).To(Equal(len(vpcList)))
+
+			accountPollerObj.selector = ces
+			vmLabelsMap := map[string]string{
```

Comment:
Lets separate out tests specific to vpc and vms.

Reply:
If we add like that there are lot of duplicate codes.

Reply:
Lets keep it this way, as all the functionality of doaccountpolling is verified here.
If this keeps growing then we can split it like you suggested.

Reply:
Solved.